src/components/SingleArticle.js

```diff
@@ -16,8 +19,29 @@ const SingleArticle = () => {
       .catch((err) => {
         setIsErr({ err });
       });
-  }, []);
+  }, [article_id]);
+  function addVote() {
+    if (!isClicked) {
+      setCount((currCount) => {
+        return currCount + 1;
+      });
+
+      patchVote(article_id, 1).then((res) => {
+        setIsClicked(true);
+      });
+    }
+  }
```

Comment:
Need some handling for then when the api request doesnt work, i.e. reseting the count back to what it was before the click.